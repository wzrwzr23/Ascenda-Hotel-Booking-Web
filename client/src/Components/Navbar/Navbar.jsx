import './Navbar.css'
import { Link } from 'react-router-dom'
import React from 'react'

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { error: null, errorInfo: null };
  }
  
  componentDidCatch(error, errorInfo) {
    // Catch errors in any components below and re-render with error message
    this.setState({
      error: error,
      errorInfo: errorInfo
    })
    // You can also log error messages to an error reporting service here
  }
  
  render() {
    if (this.state.errorInfo) {
      // Error path
      return (
        <div>
          <h2>Something went wrong.</h2>
          <details style={{ whiteSpace: 'pre-wrap' }}>
            {this.state.error && this.state.error.toString()}
            <br />
            {this.state.errorInfo.componentStack}
          </details>
        </div>
      );
    }
    // Normally, just render children
    return this.props.children;
  }  
}
const Navbar = () => {
  return (
    <div className='navbar'>
      <div className="navContainer">
        <ErrorBoundary>
        {<Link to="/" style={{ color: "inherit", textDecoration: "none" }}>
            <span className="logo">Ascenda Hotel Booking</span>
          </Link>}
        </ErrorBoundary>
        {/* <div className="navItems">
          <button className="navButtons" data-testid="navRegister">Register</button>
          <button className="navButtons">Login</button>
        </div> */}
      </div>
    </div>
  )
}

export default Navbar