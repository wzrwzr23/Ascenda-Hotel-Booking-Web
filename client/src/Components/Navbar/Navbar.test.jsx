import Navbar from './Navbar'
import React from 'react'
import ReactDOM from 'react-dom'
import { render, screen, cleanup } from '@testing-library/react';
import {jest} from '@testing-library/jest-dom'
import renderer from "react-test-renderer"

afterEach(cleanup);

it("renders without crashing", ()=>{
    const div = document.createElement("div")
    render(<Navbar></Navbar>, div)
    ReactDOM.unmountComponentAtNode(div)
})

test('renders and displays Ascenda', () => {
    render(<Navbar />);
    const linkElement = screen.getByText(/Ascenda/i);
    expect(linkElement).toBeInTheDocument();
});

it("renders with button Register", ()=>{
    render(<Navbar/>);
    const button = screen.getByTestId("navRegister");
    expect(button).toBeInTheDocument();
})

it("matches snapshot", ()=>{ //if you change the code in footer, then this test will fail. if intentional, can update snapshot. if by mistake, undo the code
    const tree = renderer.create(<Navbar/>).toJSON();
    expect(tree).toMatchSnapshot();
})

