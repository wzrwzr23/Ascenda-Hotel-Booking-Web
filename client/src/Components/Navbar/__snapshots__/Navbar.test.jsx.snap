// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="navbar"
>
  <div
    className="navContainer"
  >
    <div>
      <h2>
        Something went wrong.
      </h2>
      <details
        style={
          Object {
            "whiteSpace": "pre-wrap",
          }
        }
      >
        Error: useHref() may be used only in the context of a &lt;Router&gt; component.
        <br />
        
    at LinkWithRef (/Users/namitha.justine/Desktop/50.003/ascenda_v2/packages/react-router-dom/index.tsx:264:7)
    at ErrorBoundary (/Users/namitha.justine/Desktop/50.003/ascenda_v2/client/src/Components/Navbar/Navbar.jsx:7:5)
    at div
    at div
    at Navbar
      </details>
    </div>
    <div
      className="navItems"
    >
      <button
        className="navButtons"
        data-testid="navRegister"
      >
        Register
      </button>
      <button
        className="navButtons"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
